src/data/VariantSelector.ts

```diff
@@ -167,6 +168,49 @@ export function isValidPangoLineageQuery(query: string): boolean {
   return /^([A-Z]){1,3}(\.[0-9]{1,3})*(\.?\*)?$/.test(query.toUpperCase());
 }
 
+const stripNumber = (str: string) => Number.parseInt(str.replace(/\D+/g, ''));
+const isNumeric = (str: string) => /^\d+$/.test(str);
+
+type geneType = {
+  name: string;
+  startPosition: number;
+  endPosition: number;
+  aaSeq: string;
+};
+
+const formatGeneName = (gene: string): string => {
+  return gene.length === 1
+    ? gene.toUpperCase()
+    : gene.slice(0, -1).toUpperCase() + gene.slice(-1).toLowerCase();
+};
+
+const loadRefData = () => JSON.parse(JSON.stringify(jsonRefData));
+
+export const normalizeMutationName = (name: string) => {
+  const refData = loadRefData();
+  let items = name.split(':');
+  if (name.toLowerCase().startsWith('ins_')) {
+    return items.length === 3
+      ? `ins_${formatGeneName(items[0].substring(4))}:${items[1]}:${items[2].toUpperCase()}`
+      : `${items[0].toLowerCase()}:${items[1].toUpperCase()}`;
+  } else {
+    if (items.length === 1) {
+      if (isNumeric(name[0])) {
+        let refBase = refData.nucSeq[stripNumber(name) - 1];
+        return `${refBase}${name}`.toUpperCase();
+      }
+
+      return name.toUpperCase();
+    } else {
+      let refBase = refData.genes
+        .filter((gene: geneType) => gene.name === formatGeneName(items[0]))[0]
+        .aaSeq.toString()[stripNumber(items[1]) - 1];
+
+      return `${formatGeneName(items[0])}:${isNumeric(items[1][0]) ? refBase : ''}${items[1].toUpperCase()}`;
+    }
```

Comment:
Some documentation would also be helpful here. It's difficult to understand what all these conditions mean.

The aim of the documentation and the structure of the code would be to make it obvious to see that all relevant cases are covered.